doc/topics/development/conventions/style.rst

```diff
@@ -66,15 +56,23 @@ Formatting Strings
 
 All strings which require formatting should use the `.format` string method:
 
+Please do NOT use printf formatting, unless it's a log message.
+
+Good:
+
 .. code-block:: python
 
     data = "some text"
-    more = "{0} and then some".format(data)
+    more = "{} and then some".format(data)
+    log.debug("%s and then some", data)
 
-Make sure to use indices or identifiers in the format brackets, since empty
-brackets are not supported by python 2.6.
+Bad:
+
+.. code-block:: python
+
+    data = "some text"
+    log.debug("{} and then some".format(data))
```

Comment:
Should we create an issue for where this is done throughout Salt?

```
# Shows a lot of instances of use
grep -rn "log.debug(" | grep "\.format"
```

```
$ grep -r "log.debug(" | grep "\.format" | wc -l
97

$ grep -r "log.debug(" | grep "\.format" | grep -v "tests/" | wc -l
72
```

Reply:
There's a disabled PyLint rule which will take care of that.
The issue with enabling that is fixing the tons of bad usage across the salt code base.
We could consider that a stretch goal for Aluminum though.

Reply:
FYI, https://github.com/saltstack/salt/pull/59031